cerr/errors.go

```diff
@@ -21,154 +21,138 @@ const (
 	// Returned when a field is marked as an ID, and it's not a string
-	ErrInvalidIDType = newErr("E10035", codes.InvalidArgument, http.StatusBadRequest, "ID fields have to be strings")
+	ErrInvalidIDType = NewAsertoError("E10035", codes.InvalidArgument, http.StatusBadRequest, "ID fields have to be strings")
 	// Returned when an ID is not correct
-	ErrInvalidID = newErr("E10036", codes.InvalidArgument, http.StatusBadRequest, "invalid ID type")
+	ErrInvalidID = NewAsertoError("E10036", codes.InvalidArgument, http.StatusBadRequest, "invalid ID type")
 	// Returned when trying to delete an entity that still has dependents
-	ErrNotEmpty = newErr("E10037", codes.FailedPrecondition, http.StatusBadRequest, "entity is not empty")
+	ErrNotEmpty = NewAsertoError("E10037", codes.FailedPrecondition, http.StatusBadRequest, "entity is not empty")
 	// Returned when authentication has failed or is not possible
-	ErrAuthenticationFailed = newErr("E10038", codes.FailedPrecondition, http.StatusUnauthorized, "authentication failed")
+	ErrAuthenticationFailed = NewAsertoError("E10038", codes.FailedPrecondition, http.StatusUnauthorized, "authentication failed")
 	// Returned when a given parameter is incorrect (wrong format, value or type)
-	ErrInvalidArgument = newErr("E10039", codes.InvalidArgument, http.StatusBadRequest, "invalid argument")
+	ErrInvalidArgument = NewAsertoError("E10039", codes.InvalidArgument, http.StatusBadRequest, "invalid argument")
 	// Returned when the caller is trying to update a readonly value
-	ErrReadOnly = newErr("E10040", codes.InvalidArgument, http.StatusBadRequest, "readonly")
+	ErrReadOnly = NewAsertoError("E10040", codes.InvalidArgument, http.StatusBadRequest, "readonly")
 	// Returned when the caller tries to create or update a policy with a name that already exists
-	ErrDuplicatePolicyName = newErr("E10041", codes.InvalidArgument, http.StatusConflict, "policy name already exists")
+	ErrDuplicatePolicyName = NewAsertoError("E10041", codes.InvalidArgument, http.StatusConflict, "policy name already exists")
 	// Returned when the caller tries to create or update a connection with a name that already exists
-	ErrDuplicateConnectionName = newErr("E10042", codes.InvalidArgument, http.StatusConflict, "connection name already exists")
-	// Returned if a module is not found
-	ErrModuleNotFound = newErr("E10043", codes.NotFound, http.StatusNotFound, "module not found")
+	ErrDuplicateConnectionName = NewAsertoError("E10042", codes.InvalidArgument, http.StatusConflict, "connection name already exists")
 	// Return if a user is not found
-	ErrUserNotFound = newErr("E10044", codes.NotFound, http.StatusNotFound, "user not found")
+	ErrUserNotFound = NewAsertoError("E10044", codes.NotFound, http.StatusNotFound, "user not found")
 	// Return if a user already exists
-	ErrUserAlreadyExists = newErr("E10045", codes.AlreadyExists, http.StatusConflict, "user already exists")
```

Comment:
seems like these errors were moved to runtime: https://github.com/aserto-dev/runtime/pull/17/files#diff-75ee1df56ccde2b32c6fd863b3fe0d667818112707bda3bbba5e83e995637affR12

If someone would like to compare error codes, they would need to add a dependency to runtime. This could easily cause circular dependencies. In the long run, I think this would also cause duplicate error codes, since errors are defined in multiple places.

Would it make sense to keep all the errors together and extract them into a different package (eg. aserto-dev/aserto-errors?)